crates/miden-lib/src/utils/protocol_assembler.rs

```diff
@@ -294,9 +342,18 @@ impl ScriptBuilder {
     // TESTING CONVENIENCE FUNCTIONS
     // --------------------------------------------------------------------------------------------
 
-    /// Returns a [`ScriptBuilder`] with the `mock::{account, faucet, util}` libraries.
+    #[cfg(any(feature = "testing", test))]
+    pub fn with_kernel_library(source_manager: Arc<dyn SourceManagerSync>) -> Self {
```

Comment:
Nit: I would basically add docs which are mostly a copy of `TransactionKernel::with_kernel_library` (or link to it?)